### PR TITLE
rt refine: some cleanup

### PR DESCRIPTION
**Pull request by mktnk3**
- KHeap_R: cleaned up get_reply_valid_reply', etc.
- Reply_R, Ipc_R: some cleanup around replyPush, removing
  redundant assumptions, etc.

Signed-off-by: Miki Tanaka <miki.tanaka@data61.csiro.au>

### CODE REVIEW

**Review by MitchellBuckley — APPROVED** [review]
Very nice!! Thank you for cleaning these things up. Sorry for introducing some of that mess.

File: proof/refine/ARM/Reply_R.thy [hunk 1]
```diff
@@ -977,24 +909,37 @@ lemma updateReply_replyTCB_invs':
...
+      apply (wpsimp wp: updateReply_valid_objs')
+     apply wpsimp
+      apply (rule_tac Q="\<lambda>_. valid_objs' and reply_at' y and reply_at' replyPtr
+                         and ko_at' sc scp" in hoare_strengthen_post)
+       apply (wpsimp wp: updateReply_valid_objs')
+      apply clarsimp
+      apply (prop_tac "sc_at' scp s")
+       apply (clarsimp simp: obj_at'_def)
+      apply (clarsimp simp: valid_reply'_def valid_obj'_def
+      valid_sched_context'_def valid_sched_context_size'_def objBits_simps'
```
Comment (VictorPhan) [inline proof/refine/ARM/Reply_R.thy:935]: nitpick: indentation and line immediately after

File: proof/refine/ARM/Reply_R.thy [hunk 2]
```diff
@@ -977,24 +909,37 @@ lemma updateReply_replyTCB_invs':
...
+           dest!: sc_ko_at_valid_objs_valid_sc')+
+     apply wpsimp+
+  apply safe
+       apply ((clarsimp simp: valid_reply'_def valid_obj'_def
+       valid_sched_context'_def valid_sched_context_size'_def objBits_simps'
```
Comment (VictorPhan) [inline proof/refine/ARM/Reply_R.thy:940]: nitpick: indentation and line immediately after

**Review by VictorPhan — APPROVED** [review]
Nice cleanup

File: proof/refine/ARM/Reply_R.thy [hunk 3]
```diff
@@ -977,24 +909,37 @@ lemma updateReply_replyTCB_invs':
...
-              set_reply'.valid_objs' set_reply'.obj_at')+
-  apply (clarsimp simp: valid_obj'_def)
-  apply (drule obj_at_ko_at'[where p=scp], clarsimp)
-  apply (frule sc_ko_at_valid_objs_valid_sc', assumption)
-  apply (intro conjI impI allI;
-         clarsimp;
-         drule (1) ko_at'_inj;
-         clarsimp simp: valid_sched_context'_def valid_sched_context_size'_def objBits_def
-                        objBitsKO_def)
+ unfolding bindScReply_def
```
Comment (corlewis) [inline proof/refine/ARM/Reply_R.thy:920]: nitpick: indentation

**Review by corlewis — APPROVED** [review]
Very nice!

**Comment by mktnk3** [comment]
Err, I addressed all the indentation/spacing comments, but forgot to push that version before merging...

**Pull request merged**